lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -459,6 +468,22 @@ def add_browser_info(browser_info, post)
           userAgent: browser_info[:user_agent]
         }
       end
+
+      def failure_response(post)
+        action = 'authorise'
+        raw_response = ssl_post("#{url}/#{action}", post_data(action, post), request_headers(options))
```

Comment:
Do we have to make a second request to the gateway in order to return this failure response, after already determining that there was a failure after the initial request?

Reply:
It turns out we do not! Based on our chat yesterday, this has now been refactored to just replace the `success` and `message` parameters of the initial response without sending a new request.